cmake/LLVMIRUtil.cmake

```diff
@@ -111,6 +111,33 @@ function(llvmir_attach_bc_target)
   # compile lang flags
   llvmir_extract_lang_flags(IN_LANG_FLAGS ${LINKER_LANGUAGE})
 
+  # Link static libraries
+  if(LLVMIR_ATTACH_LINK_STATIC_LIB)
+    message(STATUS "LINK_STATIC_LIB option is turned on. Linking static libraries.")
+
+    # Find statically linked libraries, and add the source files for IR generation
+    foreach(LINK_LIB ${LINK_LIBRARIES})
+      if(TARGET ${LINK_LIB})
+        get_target_property(type ${LINK_LIB} TYPE)
+
+        # Check the library is a static one. For shared libraries, we cannot get the source files for compiling.
+        if(${type} STREQUAL "STATIC_LIBRARY")
+          get_property(LINK_LIB_FILES TARGET ${LINK_LIB} PROPERTY SOURCES)
+          message(STATUS "Linking Static library: ${LINK_LIB}")
```

Comment:
Again, adjust message accordingly:

`Attaching to dependent static Y of target X`

Reply:
Modified as suggested in the new commit